seva-web/lib/webproxy_page.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:flutter/material.dart';
+import 'websocket.dart';
+import 'dart:async';
+import 'dart:html';
+import 'dart:convert';
+import 'package:web_socket_channel/web_socket_channel.dart';
+
+void main() {
+  runApp( MaterialApp(
+       home: WebProxy()
+  ));
+}
+
+class WebProxy extends  StatefulWidget {
+  @override
+  State<WebProxy> createState() => _WebProxyState();
+}
+
+class _WebProxyState extends State<WebProxy> {
+
+  TextEditingController textarea = TextEditingController();
+  bool waiting_on_response1 = false;
+  
+  Future<WebSocketCommand> response_handler() async {
+    // catch the response code and update state accordingly
+    setState(() {
+      waiting_on_response1 = true;
+    });
+    String response = await stream.first;
+    setState(() {
+      waiting_on_response1 = false;
+    });
+    return WebSocketCommand.from_json(jsonDecode(response));
+  }
+
+  Future<void> write_proxy(String myProxy) async {
+    // writes proxy 
+    print(myProxy);
+    WebSocketCommand.outbound(myProxy, []).send();
```

Comment:
This is not the proper constructor. The first argument is the command identifier in string form (in your case "web_proxy" from here https://github.com/StaticRocket/seva/pull/1/files#diff-bf937a9b1415bf51c222fde1850f2afdee5e9002c2d13df063db9af7820e505cR59 ) the second argument is a list of strings as arguments to that command. Again, this is the foundation for a settings page, so the proper command should be something like:
```dart
WebSocketCommand.outbound("save_settings", [serialized_settings]).send();
```

Where serialized_settings is a json string like: `{"http_proxy":"http://example.com","no_proxy":"example.com"}`